sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -84,119 +84,79 @@ export class MessageSender extends LinkEntity<AwaitableSender> {
     this._retryOptions = retryOptions;
     this._onAmqpError = (context: EventContext) => {
       const senderError = context.sender && context.sender.error;
-      if (senderError) {
-        const err = translate(senderError);
-        log.error(
-          "[%s] An error occurred for sender '%s': %O.",
-          this._context.connectionId,
-          this.name,
-          err
-        );
-      }
+      log.error(
+        "[%s] 'sender_error' event occurred on the sender '%s' with address '%s'. " +
+          "The associated error is: %O",
+        this._context.connectionId,
+        this.name,
+        this.address,
+        senderError
+      );
+      // TODO: Consider rejecting promise in trySendBatch() or createBatch()
     };
 
     this._onSessionError = (context: EventContext) => {
       const sessionError = context.session && context.session.error;
-      if (sessionError) {
-        const err = translate(sessionError);
-        log.error(
-          "[%s] An error occurred on the session of sender '%s': %O.",
-          this._context.connectionId,
-          this.name,
-          err
-        );
-      }
+      log.error(
+        "[%s] 'session_error' event occurred on the session of sender '%s' with address '%s'. " +
+          "The associated error is: %O",
+        this._context.connectionId,
+        this.name,
+        this.address,
+        sessionError
+      );
+      // TODO: Consider rejecting promise in trySendBatch() or createBatch()
     };
 
     this._onAmqpClose = async (context: EventContext) => {
       const sender = this.link || context.sender!;
       const senderError = context.sender && context.sender.error;
-      if (senderError) {
-        log.error(
-          "[%s] 'sender_close' event occurred for sender '%s' with address '%s'. " +
-            "The associated error is: %O",
-          this._context.connectionId,
-          this.name,
-          this.address,
-          senderError
-        );
-      }
-      if (sender && !sender.isItselfClosed()) {
-        if (!this.isConnecting) {
-          log.error(
-            "[%s] 'sender_close' event occurred on the sender '%s' with address '%s' " +
-              "and the sdk did not initiate this. The sender is not reconnecting. Hence, calling " +
-              "detached from the _onAmqpClose() handler.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-          await this.onDetached(senderError);
-        } else {
+      log.error(
+        "[%s] 'sender_close' event occurred on the sender '%s' with address '%s'. " +
+          "Value for isItselfClosed on the receiver is: '%s' " +
+          "Value for isConnecting on the session is: '%s'.",
+        this._context.connectionId,
+        this.name,
+        this.address,
+        sender ? sender.isItselfClosed().toString() : undefined,
```

Comment:
why do we care about `isItselfClosed()`?

Reply:
I followed event-hubs here. Not sure why it was important over there.
I can change this to
```ts
      log.error(
        "[%s] 'sender_close' event occurred on the sender '%s' with address '%s'. ",
                 "The associated error is: %O",
        this._context.connectionId,
        this.name,
        this.address,
        senderError
      );
```

Reply:
Same for session_close event.

Reply:
It was used to determine whether or not we would be recovering the broken link. If the link/session was "itself closed", then it would not be recovered. We do not need to log this information anymore since we do not plan to use it

Reply:
Updated to the one I mentioned